sycl/include/sycl/ext/oneapi/experimental/annotated_arg/annotated_arg.hpp

```diff
@@ -40,6 +40,14 @@ struct HasSubscriptOperator
     : std::bool_constant<
           !std::is_void_v<decltype(std::declval<T>().operator[](0))>> {};
 
+template <class T> struct is_ann_arg_impl : std::false_type {};
+template <class T, class P>
+struct is_ann_arg_impl<annotated_arg<T, P>> : std::true_type {};
+template <class T, class P>
+struct is_ann_arg_impl<const annotated_arg<T, P>> : std::true_type {};
+template <class T>
+constexpr bool is_ann_arg_v =
+    is_ann_arg_impl<std::remove_reference_t<T>>::value;
```

Comment:
remove_ref shouldn't be here

Reply:
Why? I am still using the std::forward

Reply:
Sorry overlooked that it's named _impl. That's fine.